CRM/Contribute/BAO/Contribution.php

```diff
@@ -2953,7 +3008,19 @@ public function _assignMessageVariablesToTemplate(&$values, $input, $returnMessa
       //build an array of cId/pId of participants
       $additionalIDs = CRM_Event_BAO_Event::buildCustomProfile($this->_relatedObjects['participant']->id, NULL, $this->_relatedObjects['contact']->id, $isTest, TRUE);
       unset($additionalIDs[$this->_relatedObjects['participant']->id]);
-      //send receipt to additional participant if exists
+      $participantLine = NULL;
+
+      // Build an array of participant line items keyed my participant ID
+      if (is_array($values['lineItem'])) {
```

Comment:
The test error is happening here.

Whether that test should be creating a contribution without line items I dunno.

Reply:
@demeritcowboy I don't think the test **should** be creating a contribution without line items - that doesn't seem like behaviour that should ever come up in the running of the system.  I'm reticent to change a test based on "I don't think this should happen in the real world" though.

That aside, I've put the workaround in for the case there's no line items.  The receipt's still going to have insufficient information to fill out these fields, but it's better than the api call dying with a cryptic error.